addons/udes_stock_refactoring/models/stock_picking.py

```diff
@@ -0,0 +1,367 @@
+    )
+    def _compute_related_picking_ids(self):
+        """Compute previous/next pickings"""
+        Picking = self.env["stock.picking"]
+
+        for picking in self:
+            if picking.id:
+                picking.u_created_back_orders = Picking.get_pickings(backorder_id=picking.id)
+
+            picking.u_prev_picking_ids = picking.mapped("move_lines.move_orig_ids.picking_id")
+            picking.u_next_picking_ids = picking.mapped("move_lines.move_dest_ids.picking_id")
+
+    @api.depends("move_lines", "move_lines.move_orig_ids", "move_lines.move_orig_ids.picking_id")
+    def _compute_first_picking_ids(self):
+        """Compute first pickings"""
+        Move = self.env["stock.move"]
+
+        for picking in self:
+            first_moves = Move.browse()
+            moves = picking.move_lines
+            while moves:
+                first_moves |= moves.filtered(lambda x: not x.move_orig_ids)
+                moves = moves.mapped("move_orig_ids")
+            picking.u_first_picking_ids = first_moves.mapped("picking_id")
+
+    u_mark = fields.Boolean(
+        "Marked",
+        default=True,
+        index=True,
+        help="Pickings that are unused after refactoring are unmarked " "and deleted",
+    )
+
+    # compute previous and next pickings
+    u_prev_picking_ids = fields.One2many(
+        "stock.picking",
+        string="Previous Pickings",
+        compute="_compute_related_picking_ids",
+        help="Previous pickings",
+    )
+    u_next_picking_ids = fields.One2many(
+        "stock.picking",
+        string="Next Pickings",
+        compute="_compute_related_picking_ids",
+        help="Next pickings",
+    )
+    u_first_picking_ids = fields.One2many(
+        "stock.picking",
+        string="First Pickings",
+        compute="_compute_first_picking_ids",
+        help="First pickings in the chain",
+    )
+    u_created_back_orders = fields.One2many(
+        "stock.picking",
+        string="Created Back Orders",
+        compute="_compute_related_picking_ids",
+        help="Created Back Orders",
+    )
+
+    def find_empty_pickings_to_unlink(self):
+        """
+        Finds empty pickings to unlink in self, when it is set, otherwise
+        searches for any empty picking.
+
+        Filter out of the generic search when the picking type does not
+        have auto unlink empty enabled.
+        """
+        PickingType = self.env["stock.picking.type"]
+
+        domain = [
+            ("u_mark", "=", False),
+            ("is_locked", "=", True),
+        ]
+        if self:
+            domain.append(("id", "in", self.ids))
+        else:
+            pts = PickingType.search([("u_auto_unlink_empty", "=", False)])
+            if pts:
+                domain.append(("picking_type_id", "not in", pts.ids))
+        while True:
+            records = self.search(domain, limit=1000)
+            if not records:
+                break
+            yield records
+
+    def unlink_empty(self):
```

Comment:
It may be better to move this function to udes_stock as that is where ```u_auto_unlink_empty``` is defined and is independent of refactoring

Reply:
Also inconsistent docstring format

Reply:
Have now updated the docstrings.